### PR TITLE
Homework2: Andrey Kapitonov

### PR DESCRIPTION
**Pull request by akapitonov1999**

### CODE REVIEW

File: FizzBuzz.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# FizzBuzz task
+
+i = 0
+while i != 100:
+    i += 1
+    if i % 3 == 0 and i % 5 == 0:
```
Comment (belovmd) [inline FizzBuzz.py:6]: 15?

File: Codewars1.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+''' In this kata you will create a function
+that takes a list of non-negative integers
+and strings and returns a new list with the
+strings filtered out.'''
+
+
+def filter_list(l):
+    i = 0
```
Comment (belovmd) [inline Codewars1.py:8]: please don't use magic names l, i

File: Codewars1.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    while i != len(l):
+        if type(l[i]) != int:
+            l.pop(i)
+        else:
+            i += 1
+            continue
```
Comment (belovmd) [inline Codewars1.py:14]: we don't need continue here

File: Codewars1.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    return l
+
+
+'''An isogram is a word that has no repeating letters,
+consecutive or non-consecutive. Implement a function
+that determines whether a string that contains only letters is an isogram.
+Assume the empty string is an isogram. Ignore letter case.'''
+
+
+def is_isogram(string):
```
Comment (belovmd) [inline Codewars1.py:24]: can we just use set here?

File: Codewars1.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+to hear non-native Italian speakers trying to pronounce it :('''
+
+
+def tribonacci(signature, n):
+    if n == 0:
+        return []
+    elif n == 1:
+        listing = []
+        listing.append(signature[0])
+        return listing
```
Comment (belovmd) [inline Codewars1.py:55]: `return [signature[0]]`?

File: Codewars1.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    elif n == 2:
+        listing = []
+        listing.append(signature[0])
+        listing.append(signature[1])
+        return listing
```
Comment (belovmd) [inline Codewars1.py:60]: same as above

File: Palindrome.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        print('YES, it is an int palindrome!')
+    else:
+        print('NO, it is not a palindrome!')
+
+
+def strpalindrome(a):
+    x = ''
+    for i in a:
+        x = i + x
+        if a == x:
```
Comment (belovmd) [inline Palindrome.py:29]: just if a==a[::-1]

File: Palindrome.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+def intpalindrome(a):
+    length = int(math.log10(a)) + 1
+    i = 0
+    listing1 = []
+    while i != length:
+        x = a // (10 ** i) % 10
+        listing1.append(x)
+        i += 1
+    listing2 = listing1[::-1]
```
Comment (belovmd) [inline Palindrome.py:18]: we don't need lists here. just build reversed number using arithmetic operator

File: hello.py [hunk 9]
```diff
@@ -0,0 +1 @@
+print('Hello, Maxim!')
```
Comment (belovmd) [inline hello.py:1]: Hi!

**Review by belovmd — CHANGES_REQUESTED** [review]
Thank you for your work. 
Please address these issues:

File: Codewars1.py (hunk 2 above)
Comment (akapitonov1999) [reply Codewars1.py:8]: l is for 'list', i is for iteration

File: Codewars1.py (hunk 3 above)
Comment (akapitonov1999) [reply Codewars1.py:14]: ok

File: FizzBuzz.py (hunk 1 above)
Comment (akapitonov1999) [reply FizzBuzz.py:6]: like if it was 'if i % 15 == 0:' ?

File: Codewars1.py (hunk 2 above)
Comment (belovmd) [reply Codewars1.py:8]: better to use names that show what do they keep

**Review by belovmd — COMMENTED** [review]
7

**Pull request closed without merging**